fileio/ft_write_data.m

```diff
@@ -875,12 +875,130 @@ function ft_write_data(filename, dat, varargin)
     data.label = hdr.label;
     data.sampleinfo = [1 size(dat,2)];
     
-    % convert the raw data structure to Homer format
-    nirs = fieldtrip2homer(data, 'event', evt);
+    % devide data in nirs channels, stimulus channels and auxillary channels
+    seldat  = startsWith(hdr.chantype, 'nirs');
+    selstim = strcmp(hdr.chantype, 'stimulus');
+    selaux  = ~seldat & ~selstim;
+    
+    % create empty SnirfClass
+    snirf = SnirfClass();
+    
+    % collect information for creation of snirf file
+    source_idx = find(contains(data.hdr.opto.optotype, {'transmitter', 'source'}));
+    detector_idx = find(contains(data.hdr.opto.optotype, {'receiver', 'detector'}));
+    tra = data.hdr.opto.tra;
+    tra_t = tra'; % transpose tra matrix to get indices of wavelength by row (thus by channel)
+    wl_idx = find(tra_t>0);
+    all_wavelengths = data.hdr.opto.wavelength(tra_t(wl_idx));
+    split = median(all_wavelengths);
+    WL1.values = all_wavelengths(all_wavelengths<split);
+    WL2.values = all_wavelengths(all_wavelengths>split);
+    WL1.nominal = round(median(WL1.values),-1);
+    WL2.nominal = round(median(WL2.values),-1);
+    num_WL = 2;
+    ft_warning('assuming that the nominal wavelengths are %d and %d nm', WL1.nominal, WL2.nominal)
+     
+    % metaDataTags
+    snirf.metaDataTags(1).tags.LengthUnit = data.hdr.opto.unit; 
+    snirf.metaDataTags(1).tags.TimeUnit = 's'; 
+    snirf.metaDataTags(1).tags.FrequencyUnit = 'Hz';
+    
+    % data 
+    % (FIX ME: in theory this can contain multiple data blocks)
```

Comment:
I think this can be fixed like this

```
for trl=1:numel(data.trial)
  snirf.data(trl).dataTimeSeries = data.trial{trl}(seldat,:)';
  snirf.data(trl).time = data.time{trl}';
end
```

the same would be needed for stim and aux

Reply:
It is not entirely clear to me whether the data blocks are intended to use this way. 
In the [notes on Ted Huppert's talk](https://docs.google.com/document/d/1AqkIff1SkLmwEkYCXuwEnUrz7Wo-N7T2DKGLUGknpXQ) on snirf I found this:
"indexing for data{i} → e.g. store different kinds of data (also hybrid) (eg. frequency domain and time domain), or junks of data when your device cannot measure it continuously"
Also in the stim field it is not clear to me how stimuli in different data block would be represented, because there is only 1 stim structure for all the data blocks.

Reply:
It is also not 100% clear to me whether epoched data can be represented in SNIRF. Furthermore, in https://github.com/fNIRS/snirf/blob/master/snirf_specification.md#nirsidataj there is `/data(1)` but also `/data1`. I propose we ask somewhere in the SNIRF community.

An alternative is 

`/data(1).dataTimeSeries = cat(2, data.trial{:});`
`/data(1).data(trl).time    = cat(2, data.time{:});`

which will work fine if the data is pseudo continuous (i.e. segmented in subsequent pieces), but confusing when data is time locked to the onset of a stimulus. In that case the time axis would jump back and forth and `plot(/data(1).data(trl).time` would show a sawtooth pattern.

Reply:
I propose to leave it this way for now to make sure that no errors will happen there. I will also ask to the SNIRF community how they see it.
It might for example be that data can be divided into multiple blocks, but that the timestamps of the events always correspond to the data.time of the different blocks. This seems the most logic actually,... However, because events are linked to the data with sample numbers in FT, a lot of assumptions on how to relate the events to the data should be made here. I therefor prefer to let this up to the user and only have the option to provide one data block with one event structure.

Reply:
agreed